ipa-core/src/protocol/ipa_prf/malicious_security/prover.rs

```diff
@@ -14,15 +15,64 @@ use crate::{
     },
 };
 
-pub struct ZeroKnowledgeProof<F: PrimeField, N: ArrayLength> {
-    pub g: GenericArray<F, N>,
+pub type UVPolynomial<F, N> = (GenericArray<F, N>, GenericArray<F, N>);
+pub type TwoNMinusOne<R> = Diff<Sum<R, R>, U1>;
+
+// todo: deprecate
+pub type TwoNPlusOne<R> = Sum<Sum<R, R>, U1>;
+
+/// The purpose of this trait is to define an associated type,
+/// i.e. the length of the array `Length`
+pub trait ProofArray {
+    type Length: ArrayLength;
+}
+
+/// This struct contains a single `ZeroKnowledgeProof`
+/// The length of the proof is not the generic parameter `R`.
+/// The length can be obtained using the trait `ProofArray`
+/// where the associated type `Length`
+/// represents the length of the proof.
+///
+/// Parameter `R` represents the recursion factor, which is a parameter
+/// used during the proof generation and verification.
```

Comment:
I was planning to change verifier.rs as well as soon as I come to the verification protocol. I changed it to `R` mainly because λ doesn't fit the norm to use capital case letter for generics. I ended up putting 
`#[allow(non_camel_case_types)]` everywhere. λ is also annoying to type.